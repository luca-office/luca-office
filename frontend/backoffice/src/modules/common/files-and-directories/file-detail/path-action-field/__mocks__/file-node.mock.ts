import {FileNode} from "shared/components"
import {BinaryType, NodeType} from "shared/enums"

export const fileNodeMock: FileNode = {
  binaryFileId: "spfosdkfsd",
  binaryFileUrl: "url@url.de",
  binaryType: BinaryType.Image,
  id: "sfpoksdf-fposjfd",
  name: "test-file",
  parentId: "sfopksdf-fsdpfosdk",
  tags: [],
  type: NodeType.File
}
