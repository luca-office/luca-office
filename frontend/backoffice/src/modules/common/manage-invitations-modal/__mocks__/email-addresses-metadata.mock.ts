import {EmailAddressesMetadata} from "../models/email-addresses-metadata"

export const emailAddressesMetadataMock: EmailAddressesMetadata = {
  validAddresses: [
    "testmail@test.de",
    "cap@cap3.de",
    "testmail@test.de",
    "testmai2l@test.de",
    "cap3@cap3.de",
    "cap@cap3.de"
  ],
  uniqueAddresses: ["testmail@test.de", "cap@cap3.de", "testmai2l@test.de", "cap3@cap3.de"],
  hasInvalidAddress: false,
  duplicatedAddresses: ["testmail@test.de"],
  alreadyInvitedAddresses: ["cap@cap3.de"]
}
