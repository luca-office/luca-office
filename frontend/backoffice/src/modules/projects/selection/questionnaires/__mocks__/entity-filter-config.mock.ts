import {EntityFilterOptionState, EntityFilterOptionType, EntitySortOption} from "../../../../../enums"
import {EntityFilterConfig} from "../../../../../models"

export const entityFilterConfigMock: EntityFilterConfig = {
  sortBy: EntitySortOption.BY_NAME,
  state: EntityFilterOptionState.SHOW_ALL,
  type: EntityFilterOptionType.ALL_ENTITIES,
  search: ""
}
