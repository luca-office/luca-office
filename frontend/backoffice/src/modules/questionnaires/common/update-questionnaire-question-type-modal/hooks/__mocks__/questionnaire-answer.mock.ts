import {QuestionnaireAnswer} from "shared/models"

export const questionnaireAnswerMock = (
  questionId: UUID = "18316dac-2355-4e45-a7f3-f339461a488f"
): QuestionnaireAnswer => ({
  __typename: "QuestionnaireAnswer",
  createdAt: new Date(2020, 10, 5).toISOString(),
  modifiedAt: new Date(2020, 10, 15).toISOString(),
  id: "1fce8289-71fa-4568-bccc-329dc4cc67eb",
  text: "answer",
  questionId,
  isCorrect: false,
  position: 1
})
