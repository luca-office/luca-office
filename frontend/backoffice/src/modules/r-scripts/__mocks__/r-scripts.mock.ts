import {RScript} from "shared/models"

export const rScriptMock: RScript = {
  __typename: "RScript",
  id: "18de306c-a8f8-4311-9a2f-ca5418a245bd",
  createdAt: "2020-08-03 15:16:28.501000 +00:00",
  title: "title",
  description: "description",
  modifiedAt: "2020-08-03 15:16:28.501000 +00:00",
  archivedAt: null,
  gitCommitHash: "sd",
  version: "1.2",
  isEditable: true
}

export const rScriptsMock: RScript[] = [
  rScriptMock,
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2452d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2453d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2454d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2455d"}
]
