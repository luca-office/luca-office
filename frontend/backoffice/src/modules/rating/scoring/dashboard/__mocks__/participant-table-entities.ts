import {ParticipantTableEntity} from "../scoring-dashboard-table/scoring-dashboard-table"

export const mockParticipantTableEntities: ParticipantTableEntity[] = [
  {
    id: "1",
    isFinalScore: false,
    isRatingCompleted: false,
    isRatingOfMainRater: false,
    name: "name 1",
    index: 1,
    ratingCounts: {ratableProjectModulesCount: 12, finalRatedProjectModulesCount: 6},
    isRatingOfAllModulesPossible: true,
    isRatingOfSomeModulePossible: true
  },
  {
    id: "2",
    isFinalScore: false,
    isRatingCompleted: false,
    isRatingOfMainRater: false,
    name: "name 3",
    index: 2,
    ratingCounts: {ratableProjectModulesCount: 12, finalRatedProjectModulesCount: 6},
    isRatingOfAllModulesPossible: true,
    isRatingOfSomeModulePossible: true
  },
  {
    id: "3",
    isFinalScore: false,
    isRatingCompleted: false,
    isRatingOfMainRater: false,
    name: "name 3",
    index: 3,
    ratingCounts: {ratableProjectModulesCount: 12, finalRatedProjectModulesCount: 6},
    isRatingOfAllModulesPossible: true,
    isRatingOfSomeModulePossible: true
  }
]
