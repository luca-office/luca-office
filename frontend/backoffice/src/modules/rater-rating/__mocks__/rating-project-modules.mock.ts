import {projectModulesMockWithQuestionnaire, surveyInvitationsMock} from "shared/graphql/__mocks__"
import {RatingProjectModule} from "../models"

export const ratingProjectModulesMock: RatingProjectModule[] = projectModulesMockWithQuestionnaire.map(
  (projectModule, index) => ({
    ...projectModule,
    ratingsByParticipants: surveyInvitationsMock.map(({id}, index) => ({
      participantId: id,
      isFullyRated: index % 2 === 0,
      isNotRatable: false
    })),
    participantCount: index + 1,
    ratedParticipantCount: index
  })
)
