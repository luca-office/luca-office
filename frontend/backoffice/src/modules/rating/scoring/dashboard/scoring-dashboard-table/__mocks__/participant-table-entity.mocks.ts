import {surveyInvitationsMock} from "shared/graphql/__mocks__"
import {ParticipantTableEntity} from "../scoring-dashboard-table"

export const participantTableEntityMocks: ParticipantTableEntity[] = surveyInvitationsMock.map(
  (surveyInvitation, index) => ({
    id: surveyInvitation.id,
    name: surveyInvitation.participantData
      ? `${surveyInvitation.participantData.firstName} ${surveyInvitation.participantData.lastName}`
      : surveyInvitation.token,
    index: index + 1,
    isFinalScore: index === 0,
    isRatingCompleted: index === 1,
    isRatingOfMainRater: index === 2,
    ratingCounts: {ratableProjectModulesCount: 12, finalRatedProjectModulesCount: 6},
    isRatingOfAllModulesPossible: true,
    isRatingOfSomeModulePossible: true
  })
)
