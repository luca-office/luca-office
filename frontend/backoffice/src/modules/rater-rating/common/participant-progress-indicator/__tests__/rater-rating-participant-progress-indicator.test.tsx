import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Text} from "shared/components"
import {surveyInvitationsMock} from "shared/graphql/__mocks__"
import {ratingProjectModulesMock} from "../../../__mocks__/rating-project-modules.mock"
import {
  RaterRatingParticipantProgressIndicator,
  RaterRatingParticipantProgressIndicatorProps
} from "../rater-rating-participant-progress-indicator"

const defaultProps: RaterRatingParticipantProgressIndicatorProps = {
  projectModules: ratingProjectModulesMock,
  surveyInvitation: surveyInvitationsMock[0]
}

const getComponent = (props?: Partial<RaterRatingParticipantProgressIndicatorProps>) => (
  <RaterRatingParticipantProgressIndicator {...{...defaultProps, ...props}} />
)

describe("rater-rating-participant-progress-indicator", () => {
  it("renders correctly", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure", () => {
    const component = getComponent()
    const tree = shallow(component).dive()

    expect(tree.find(Text)).toHaveLength(1)
    expect(tree.find(".completed-indicator")).toHaveLength(4)
    expect(tree.find(".in-progress-indicator")).toHaveLength(0)
    expect(tree.find(".open-indicator")).toHaveLength(0)
  })
})
