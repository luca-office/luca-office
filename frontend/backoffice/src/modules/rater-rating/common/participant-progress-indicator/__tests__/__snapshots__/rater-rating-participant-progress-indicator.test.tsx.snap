// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rater-rating-participant-progress-indicator renders correctly 1`] = `
<div
  className="css-ij7jn1-ModuleProgressIndicator"
>
  <div
    className="css-17zbm7y-Text"
  >
    4/4
  </div>
  <div
    className="css-1xf60qh-ModuleProgressIndicator"
  >
    <div
      className="completed-indicator css-c39ufe-ModuleProgressIndicator"
    />
    <div
      className="completed-indicator css-c39ufe-ModuleProgressIndicator"
    />
    <div
      className="completed-indicator css-c39ufe-ModuleProgressIndicator"
    />
    <div
      className="completed-indicator css-c39ufe-ModuleProgressIndicator"
    />
  </div>
</div>
`;
